src/System.Diagnostics.Process/src/System/Diagnostics/Process.FreeBSD.cs

```diff
@@ -0,0 +1,192 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using System.ComponentModel;
+
+namespace System.Diagnostics
+{
+    public partial class Process
+    {
+        /// <summary>
+        /// Creates an array of <see cref="Process"/> components that are associated with process resources on a
+        /// remote computer. These process resources share the specified process name.
+        /// </summary>
+        public static Process[] GetProcessesByName(string processName, string machineName)
+        {
+            if (processName == null)
+            {
+                processName = string.Empty;
+            }
+
+            Process[] procs = GetProcesses(machineName);
+            var list = new List<Process>();
+
+            for (int i = 0; i < procs.Length; i++)
+            {
+                if (string.Equals(processName, procs[i].ProcessName, StringComparison.OrdinalIgnoreCase))
+                {
+                    list.Add(procs[i]);
+                }
+                else
+                {
+                    procs[i].Dispose();
+                }
+            }
+
+            return list.ToArray();
+        }
```

Comment:
This looks identical to the .OSX.cs implementation.  Do they need to be duplicated, or could we have code that's shared between macOS and FreeBSD be in .BSD.cs files that are compiled into both builds, and only have code here that's unique to FreeBSD?

Same goes for other duplication.

Reply:
NetBSD and FreeBSD are much more closer one with the other than BSD and Darwin. Ideally there would be used `#ifdefs`-like switches for OS-specific nits when needed.